airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/java/io/airbyte/cdk/extensions/LoggingInvocationInterceptor.java

```diff
@@ -0,0 +1,150 @@
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.junit.jupiter.api.extension.DynamicTestInvocationContext;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This allows us to have junit loglines in our test logs. This is instanciated via Java's
+ * ServiceLoader
+ * (https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/util/ServiceLoader.html) The
+ * declaration can be found in resources/META-INF/services/org.junit.jupiter.api.extension.Extension
+ */
+public class LoggingInvocationInterceptor implements InvocationInterceptor {
+
+  private static final class LoggingInvocationInterceptorHandler implements InvocationHandler {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(LoggingInvocationInterceptor.class);
+
+    @Override
+    public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+      if (LoggingInvocationInterceptor.class.getDeclaredMethod(method.getName(), Invocation.class, ReflectiveInvocationContext.class,
+          ExtensionContext.class) == null) {
+        LOGGER.error("Junit LoggingInvocationInterceptor executing unkown interception point " + method.getName());
+        return method.invoke(proxy, args);
+      }
+      Invocation<?> invocation = (Invocation<?>) args[0];
+      ReflectiveInvocationContext<Method> invocationContext = (ReflectiveInvocationContext<Method>) args[1];
+      ExtensionContext extensionContext = (ExtensionContext) args[2];
+      String methodName = method.getName();
+      String logLineSuffix;
+      if (methodName.equals("interceptDynamicTest")) {
+        logLineSuffix = "execution of DynamicTest " + extensionContext.getDisplayName();
+      } else if (methodName.equals("interceptTestClassConstructor")) {
+        logLineSuffix = "instance creation for " + invocationContext.getTargetClass();
+      } else if (methodName.startsWith("intercept") && methodName.endsWith("Method")) {
+        String interceptedEvent = methodName.substring("intercept".length(), methodName.length() - "Method".length());
+        logLineSuffix = "execution of @" + interceptedEvent + " method " + invocationContext.getExecutable().getDeclaringClass().getSimpleName()
+            + "." + invocationContext.getExecutable().getName();
+      } else {
+        logLineSuffix = "execution of unknown intercepted call " + methodName;
+      }
+      LOGGER.info("Junit starting " + logLineSuffix);
+      try {
+        long start = System.nanoTime();
+        Object retVal = invocation.proceed();
+        long elapsedMs = (System.nanoTime() - start) / 1_000_000;
```

Comment:
nit: Is there a reason to not use the `java.time` library? I don't care too much, but since this isn't idiomatic I'm wondering if there is.